### PR TITLE
Update build scripts

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] ~Are tests passing locally?~
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [ ] ~Did you check UI tests on the sample app? They are not executed on CI.~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

1. Add build Catalyst for XCFramework.
2. Change output directories for all platforms.
3. Add build for both mode (Debug and Release).
4. Update XCFramework scheme.

## Related PRs or issues

[AB#80352](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/80352)

## Misc

Release mode is required to build Sasquatch applications that depend on fat frameworks.

### CODE REVIEW

File: Scripts/build-tvos-framework.sh [hunk 1]
```diff
@@ -11,41 +11,44 @@ echo "Building ${TARGET_NAME}."
...
+# Building both architectures.
+xcodebuild -project "${PROJECT_NAME}.xcodeproj" -configuration "${CONFIGURATION}" -target "${TARGET_NAME}" -sdk appletvos CONFIGURATION_BUILD_DIR="${OUTPUT_DEVICE_DIR}"
+xcodebuild -project "${PROJECT_NAME}.xcodeproj" -configuration "${CONFIGURATION}" -target "${TARGET_NAME}" -sdk appletvsimulator CONFIGURATION_BUILD_DIR="${OUTPUT_SIMULATOR_DIR}"
+
+# Copy framework.
+cp -R "${OUTPUT_DEVICE_DIR}/${PROJECT_NAME}.framework" "${PRODUCTS_DIR}"
 
-# Uses the Lipo Tool to merge both binary files (i386/x86_64 + armv7/armv7s/arm64) into one Universal final product.
-lipo -create "${DEVICE_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" "${SIMULATOR_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" -output "${PRODUCTS_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}"
+# # Uses the Lipo Tool to merge both binary files (i386/x86_64 + armv7/armv7s/arm64) into one Universal final product.
```
Comment (vvechkanov) [inline Scripts/build-tvos-framework.sh:53]:
```suggestion
# Uses the Lipo Tool to merge both binary files (i386/x86_64 + armv7/armv7s/arm64) into one Universal final product.
```

File: Scripts/build-ios-framework.sh [hunk 2]
```diff
@@ -21,57 +21,60 @@ echo "Building ${TARGET_NAME}."
 # Install dir will be the final output to the framework.
 # The following line create it in the root folder of the current project.
 PRODUCTS_DIR="${SRCROOT}/../AppCenter-SDK-Apple/iOS"
-
-# The directory to gather all frameworks and build it into xcframework.
-XCFRAMEWORK_DIR="${SRCROOT}/../AppCenter-SDK-Apple/xcframework"
+OUTPUT_DIR="${SRCROOT}/../AppCenter-SDK-Apple/output"
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:24]: Why do you need this dir?

File: Scripts/build-ios-framework.sh [hunk 3]
```diff
@@ -21,57 +21,60 @@ echo "Building ${TARGET_NAME}."
...
+fi
 
 # Creates and renews the final product folder.
 mkdir -p "${PRODUCTS_DIR}"
 
-# Copy framework.
-cp -R "${DEVICE_DIR}/${PROJECT_NAME}.framework" "${PRODUCTS_DIR}"
+# Create temp directories.
+mkdir -p "${OUTPUT_DEVICE_DIR}"
+mkdir -p "${OUTPUT_SIMULATOR_DIR}"
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:50]: Do we reeally need to create it manually?

File: Scripts/build-ios-framework.sh [hunk 4]
```diff
@@ -21,57 +21,60 @@ echo "Building ${TARGET_NAME}."
...
   echo "Use current Xcode version and lipo -create the fat binary."
-  lipo -create "${DEVICE_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" "${SIMULATOR_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" -output "${PRODUCTS_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}"
+  lipo -create "${OUTPUT_DEVICE_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" "${OUTPUT_SIMULATOR_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" -output "${PRODUCTS_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}"
 else
 
-  # Grep the output of `lipo -archs` if it contains "arm64e". If it does, don't build for arm64e again.
+# Grep the output of `lipo -archs` if it contains "arm64e". If it does, don't build for arm64e again.
 DOES_CONTAIN_ARM64E=`env DEVELOPER_DIR="$MS_ARM64E_XCODE_PATH" /usr/bin/lipo -archs "${LIB_IPHONEOS_FINAL}" | grep arm64e`
 
 if [ ! -z "${DOES_CONTAIN_ARM64E}" ]; then
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:80]: Xcode 11 is a minimal version now, please remove this workaround

File: Scripts/build-macos-framework.sh [hunk 5]
```diff
@@ -11,36 +11,30 @@ echo "Building ${TARGET_NAME}."
...
   rm -rf "${PRODUCTS_DIR}/${PROJECT_NAME}.framework"
 fi
-rm -rf "${XCFRAMEWORK_RELEASE_DIR}/${PROJECT_NAME}.framework"
+if [ -d "${OUTPUT_DEVICE_DIR}/${PROJECT_NAME}.framework" ]; then
+  rm -rf "${OUTPUT_DEVICE_DIR}/${PROJECT_NAME}.framework"
+fi
 
 # Creates and renews the final product folder.
 mkdir -p "${PRODUCTS_DIR}"
+mkdir -p "${OUTPUT_DEVICE_DIR}"
```
Comment (MatkovIvan) [inline Scripts/build-macos-framework.sh:33]: Same

File: Scripts/build-ios-framework.sh [hunk 6]
```diff
@@ -21,57 +21,60 @@ echo "Building ${TARGET_NAME}."
...
+# Cleaning the previous builds.
 if [ -d "${PRODUCTS_DIR}/${PROJECT_NAME}.framework" ]; then
   rm -rf "${PRODUCTS_DIR}/${PROJECT_NAME}.framework"
 fi
+if [ -d "${OUTPUT_DEVICE_DIR}/${PROJECT_NAME}.framework" ]; then
+  rm -rf "${OUTPUT_DEVICE_DIR}/${PROJECT_NAME}.framework"
+fi
+if [ -d "${OUTPUT_SIMULATOR_DIR}/${PROJECT_NAME}.framework" ]; then
+  rm -rf "${OUTPUT_SIMULATOR_DIR}/${PROJECT_NAME}.framework"
+fi
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:43]: Why do you need it? "clean" should do the job

File: Scripts/build-xcframework.sh [hunk 7]
```diff
@@ -4,26 +4,28 @@
...
+# Create a command to build XCFramework.
+for SDK in iphoneos iphonesimulator appletvos appletvsimulator macOS maccatalyst; do
+    FRAMEWORK_PATH="${ROOT_DIR}/output/${CONFIGURATION}-${SDK}/${PROJECT_NAME}.framework"
+    [ -e "${FRAMEWORK_PATH}" ] && XC_BUILD_COMMAND="${XC_BUILD_COMMAND} -framework ${FRAMEWORK_PATH}";
+done
+XC_BUILD_COMMAND="xcodebuild -create-xcframework ${XC_BUILD_COMMAND} -output ${XC_FRAMEWORK_PATH}"
 
-XC_BUILD_COMMAND="xcodebuild -create-xcframework $XC_BUILD_COMMAND -output $XC_FRAMEWORK_PATH"
+#Build XCFramework
 eval "$XC_BUILD_COMMAND"
```
Comment (MatkovIvan) [inline Scripts/build-xcframework.sh:31]: No need to "eval" here

File: Scripts/build-xcframework.sh [hunk 8]
```diff
@@ -4,26 +4,28 @@
...
-# Create a cycle instead next lines
-SetXcBuildCommandFramework "iphoneos"
-SetXcBuildCommandFramework "iphonesimulator"
-SetXcBuildCommandFramework "appletvos"
-SetXcBuildCommandFramework "appletvsimulator"
-SetXcBuildCommandFramework "macos"
+# Create a command to build XCFramework.
+for SDK in iphoneos iphonesimulator appletvos appletvsimulator macOS maccatalyst; do
+    FRAMEWORK_PATH="${ROOT_DIR}/output/${CONFIGURATION}-${SDK}/${PROJECT_NAME}.framework"
+    [ -e "${FRAMEWORK_PATH}" ] && XC_BUILD_COMMAND="${XC_BUILD_COMMAND} -framework ${FRAMEWORK_PATH}";
```
Comment (MatkovIvan) [inline Scripts/build-xcframework.sh:26]: Remove check. It must be there

File: Scripts/build-ios-framework.sh [hunk 9]
```diff
@@ -21,57 +21,60 @@ echo "Building ${TARGET_NAME}."
...
 
-mkdir -p "${XCFRAMEWORK_DIR}"
+# Clean build.
+xcodebuild -project "${PROJECT_NAME}.xcodeproj" -configuration "${CONFIGURATION}" -target "${TARGET_NAME}" clean
 
-# Copy all framework files to use them for xcframework file creation.
-cp -R "${WORK_DIR}/" "${XCFRAMEWORK_DIR}"
+# Building both architectures.
+xcodebuild -project "${PROJECT_NAME}.xcodeproj" -configuration "${CONFIGURATION}" -target "${TARGET_NAME}" -sdk iphoneos CONFIGURATION_BUILD_DIR="${OUTPUT_DEVICE_DIR}"
+xcodebuild -project "${PROJECT_NAME}.xcodeproj" -configuration "${CONFIGURATION}" -target "${TARGET_NAME}" -sdk iphonesimulator CONFIGURATION_BUILD_DIR="${OUTPUT_SIMULATOR_DIR}"
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:57]: Remove `CONFIGURATION_BUILD_DIR="${OUTPUT_DEVICE_DIR}"`, use default setting for it

File: Scripts/build-docs.sh [hunk 10]
```diff
@@ -21,4 +21,4 @@ if [ ! -d "${DOCUMENTATION_DIR}" ]; then
 fi
 
 # Copy generated documentation into the documentation folder.
-cp -R "${SRCROOT}/../Documentation/${OS_NAME}/${PROJECT_NAME}/Generated/" "${DOCUMENTATION_DIR}"
+cp -R "${SRCROOT}/../Documentation/${OS_NAME}/${PROJECT_NAME}/Generated/" "${DOCUMENTATION_DIR}"
```
Comment (MatkovIvan) [inline Scripts/build-docs.sh:24]: Revert changes in this file

File: Scripts/build-ios-framework.sh [hunk 11]
```diff
@@ -22,77 +14,37 @@ echo "Building ${TARGET_NAME}."
...
+build() {
+    # Print only target name and issues. Mimic Xcode output to make prettify tools happy.
+    echo "=== BUILD TARGET $1 OF PROJECT ${PROJECT_NAME} WITH CONFIGURATION ${CONFIGURATION} ==="
+    # OBJROOT must be customized to avoid conflicts with the current process.
+    xcodebuild -quiet \
+        SYMROOT="${SYMROOT}" OBJROOT="${BUILT_PRODUCTS_DIR}" PROJECT_TEMP_DIR="${PROJECT_TEMP_DIR}" ONLY_ACTIVE_ARCH=NO \
+        -project "${SRCROOT}/${PROJECT_NAME}.xcodeproj" -configuration "${CONFIGURATION}" -target "$1" -sdk "$2"
+}
+build "${TARGET_NAME}" iphoneos
+build "${TARGET_NAME}" iphonesimulator
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:39]: Move it above cleaning like before

File: Scripts/build-ios-framework.sh [hunk 12]
```diff
@@ -22,77 +14,37 @@ echo "Building ${TARGET_NAME}."
...
-    # Lipo the binaries that were built with various Xcode versions.
-    env DEVELOPER_DIR="${MS_ARM64E_XCODE_PATH}" lipo -create "${DEVICE_TEMP_DIR}/${PROJECT_NAME}" "${DEVICE_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" -output "${PRODUCTS_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}"
-  fi
-
-  echo "Use arm64e Xcode and lipo -create the fat binary."
-  env DEVELOPER_DIR="$MS_ARM64E_XCODE_PATH" lipo -create "${DEVICE_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" "${SIMULATOR_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" -output "${PRODUCTS_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}"
-
-# End of arm64e code block.
-fi
+lipo -create "${OUTPUT_DEVICE_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" "${OUTPUT_SIMULATOR_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}" -output "${PRODUCTS_DIR}/${PROJECT_NAME}.framework/${PROJECT_NAME}"
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:50]: Add comment

**Comment by annakocheshkova** [comment]
I have this warning when I run **All XCFrameworks**. You have to add a check for Distribute in build-xcframework script.
image

File: Scripts/build-xcframework.sh [hunk 13]
```diff
@@ -4,26 +4,33 @@
 # Licensed under the MIT License.
 
 # Work dir is directory where all XCFramework artifacts is stored.
-WORK_DIR="${SRCROOT}/../AppCenter-SDK-Apple/xcframework"
+PRODUCTS_DIR="${SRCROOT}/../AppCenter-SDK-Apple/${ROOT_DIR}/XCFramework"
```
Comment (MatkovIvan) [inline Scripts/build-xcframework.sh:7]:
```suggestion
PRODUCTS_DIR="${SRCROOT}/../AppCenter-SDK-Apple/XCFramework"
```

**Comment by annakocheshkova** [comment]
Also, please clean up the existing xcframework before building
image

File: AppCenter.xcworkspace/xcshareddata/xcschemes/All iOS Documentation+Frameworks.xcscheme [hunk 14]
```diff
@@ -3,7 +3,7 @@
    LastUpgradeVersion = "1010"
    version = "1.3">
    <BuildAction
-      parallelizeBuildables = "YES"
+      parallelizeBuildables = "NO"
```
Comment (MatkovIvan) [inline AppCenter.xcworkspace/xcshareddata/xcschemes/All iOS Documentation+Frameworks.xcscheme:6]: Is it nessasary?

**Pull request merged**